add RavenDB integration (#2960)
DataDog/integrations-extras: ravendb/README.md

# Agent Check: RavenDB

![RavenDB Dashboard](images/ravendb_overview.png)

## Overview

> [RavenDB][1] is a NoSQL document-oriented database designed for high performance, scalability, and ease of use. It is optimized for handling JSON-based documents and supports advanced features like ACID transactions, full-text search, and automatic indexing. 

Reviewer: ```suggestion
[RavenDB][1] is a NoSQL document-oriented database designed for high performance, scalability, and ease of use. It is optimized for handling JSON-based documents and supports advanced features such as ACID transactions, full-text search, and automatic indexing. 
```

Author: done

RavenDB is widely used for applications that require high-speed transactions, real-time analytics, and distributed architecture while maintaining data integrity. It is particularly suited for enterprise applications, e-commerce, and real-time monitoring systems.

This integration collects metrics from RavenDB via its Prometheus/OpenMetrics endpoint, enabling visibility into cluster health, node activity, database operations, indexing performance, and resource utilization. These metrics can be used for monitoring, visualization, and alerting within Datadog.

## Setup

### Prerequisites

- [A running RavenDB server (single node or cluster)](https://docs.ravendb.net/start/getting-started)

> - [A valid RavenDB license](https://ravendb.net/buy)

Reviewer: Can you move these links to the bottom of the page and use reference link format?

Author: done

- The Datadog Agent installed on each node you want to monitor
- Network access from the Datadog Agent to the RavenDB metrics endpoint

### Installation

To install the RavenDB check on your host:

1. Install the [developer toolkit]

> (https://docs.datadoghq.com/developers/integrations/python/)

Reviewer: Can you format this as a reference link at the bottom of the file?

```suggestion
1. Install the [developer toolkit](https://docs.datadoghq.com/developers/integrations/python/)
```

Author: done

 on any machine.

2. Run `ddev release build ravendb` to build the package.

3. [Download the Datadog Agent][2].

4. Upload the build artifact to any host with an Agent and
 run `datadog-agent integration install -w
 path/to/ravendb/dist/<ARTIFACT_NAME>.whl`.

### Configuration

#### Metric collection

> 1. RavenDB is exposing metrics in [Prometheus/OpenMetrics](https://docs.ravendb.net/server/administration/monitoring/prometheus/) format  at an administrative endpoint: `/admin/monitoring/v1/prometheus`.

Reviewer: ```suggestion
1. RavenDB exposes metrics in [Prometheus/OpenMetrics](https://docs.ravendb.net/server/administration/monitoring/prometheus/) format  at an administrative endpoint: `/admin/monitoring/v1/prometheus`.
```

Author: done

> 2. Edit the `ravendb/conf.yaml` file in the `conf.d/` directory of the Datadog Agent. See the [sample ravendb/conf.yaml][4] for all available configuration options.

Reviewer: ```suggestion
2. Edit the `ravendb/conf.yaml` file in the `conf.d/` directory of the Datadog Agent. For all available configuration option, see the [sample ravendb/conf.yaml][4].
```

Author: done

```yaml
   instances:
     - openmetrics_endpoint: "<RAVENDB_METRICS_URL>"
       enable_database_metrics: true
       enable_index_metrics: true
       enable_collection_metrics: true
       enable_gc_metrics: true
```

#### Secured endpoints (TLS)

If the RavenDB metrics endpoint is exposed over HTTPS and requires client authentication, configure TLS options:
```yaml
instances:
  - openmetrics_endpoint: "https://<host>/admin/monitoring/v1/prometheus"
    tls_cert: /path/to/client.crt
    tls_private_key: /path/to/client.key
```
Ensure the Datadog Agent user has read access to the certificate and key files.

#### Cluster deployments

For RavenDB clusters:

- Install and configure the Datadog Agent on each node
- Configure each Agent instance to scrape the local node's metrics endpoint
- Use tags to correlate metrics across nodes and clusters

Example:
```yaml
instances:
  - openmetrics_endpoint: "<RAVENDB_METRICS_URL>"
    tags:
      - ravendb_cluster:my-ravendb-cluster
      - ravendb_node:A
```
#### Log collection (optional)

To collect RavenDB logs, enable log collection in datadog.yaml:

> logs_enabled: true

Reviewer: ````suggestion
```yaml
logs_enabled: true
```
````

Author: done

Then add a log configuration to ravendb.d/conf.yaml:
```yaml
logs:
  - type: file
    path: /var/log/ravendb/*.log
    service: ravendb
    source: ravendb

    log_processing_rules:
      - type: multi_line
        name: ravendb_multiline
        pattern: '^\d{4}-\d{2}-\d{2}'

```
Ensure the Datadog Agent user has read permissions on the log files.

### Validation

> Run the Agent's status subcommand and look for 'ravendb' under the Checks section.

Reviewer: ```suggestion
Run the [Agent's status subcommand][6] and look for `ravendb` under the Checks section.
```

Author: done

> ## Data Collected

Reviewer: This is missing a **Logs** subsection describing what log types are collected, and a **Service checks** subsection linking to `service_checks.json`.

Author: done

> ## Data Collected

Reviewer: ```suggestion
## Data collected
```

Author: done

### Metrics

See [metadata.csv][7] for a list of metrics provided by this integration.

### Events

The RavenDB integration does not include any events.

## Troubleshooting

> Need help? Contact the support@ravendb.net of this integration.

Reviewer: ```suggestion
Need help? Contact [RavenDB's support team][10].
```

Author: done

[1]: https://ravendb.net
[2]: https://app.datadoghq.com/account/settings/agent/latest

> [3]: https://docs.datadoghq.com/containers/kubernetes/integrations/

Reviewer: This isn't referenced on the page.

```suggestion
```

Author: done

[4]: https://github.com/DataDog/integrations-extras/blob/master/ravendb/datadog_checks/ravendb/data/conf.yaml.example

> [5]: https://docs.datadoghq.com/agent/configuration/agent-commands/#start-stop-and-restart-the-agent

Reviewer: This isn't referenced on the page.

```suggestion
```

Author: done

[6]: https://docs.datadoghq.com/agent/configuration/agent-commands/#agent-status-and-information
[7]: https://github.com/DataDog/integrations-extras/blob/master/ravendb/metadata.csv

> [9]: https://docs.datadoghq.com/help/

Reviewer: This isn't referenced on the page.

```suggestion
```

Author: done

> 

Reviewer: ```suggestion
[10]: mailto:support@ravendb.net
```

Author: done
